lib/staccato/v4.rb

```diff
@@ -1,58 +1,56 @@
+# frozen_string_literal: true
+
 require 'json'
+require_relative 'v4/adapter_defaults'
+require_relative 'v4/event'
+require_relative 'v4/tracker'
 
 module Staccato
+
+  # The namespace for Google Analytics V4 classes and modules.
+  #
+  # V4 utility methods are also defined here.
   module V4
+    extend AdapterDefaults
+
     # Build a new tracker instance
-    #   If the first argument is explicitly `nil`, a `NoopTracker` is returned
-    #   which responds to all the same `tracker` methods but does no tracking
+    #   If measurement_id, api_secret, or client_id are explicitly `nil`, a
+    #   `NoopTracker` is returned which responds to all the same `tracker`
+    #   methods but does no tracking
+    #
+    # (see Tracker#initialize)
     #
-    # @param measurement_id [String] the required id provided by google, i.e., `G-XXXXXXXX`
-    # @param client_id [String, Integer, nil] a unique id to track the session of
-    #   an individual user
-    # @param options [Hash] options for use in all hits from this tracker
     # @yield [Staccato::Tracker] the new tracker
     # @return [Staccato::Tracker] a new tracker is returned
-    def self.tracker(measurement_id, api_secret, client_id = nil, options = {})
-      klass = measurement_id.nil? ? Staccato::V4::NoopTracker : Staccato::V4::Tracker
+    def self.tracker(measurement_id, api_secret, client_id, **options)
```

Comment:
Can you share the benefit to doing `**options` here? Is it because you've got explicit named args in the tracker class?

Reply:
Using `**kwargs` vs `options = {}` behaves differently in ruby 2.0-2.6 vs 3.0+ (with 2.7 splitting the difference).  It's is backward's compatible to use faux-kwargs when calling a method that uses `options = {}` in its definition.  The difference is _especially_ noticeable when you precede the kwargs with optional params, but I changed `client_id` to be required.

So, I think the main reason for making this change right _here_ is that `**options` is a better semantic match.  When I see kwargs syntax in rdoc (or yard) it _explicitly_ says to me "this gobbles up all remaining keyword arguments (and probably forwards)" vs "the fourth positional argument is an options object, probably a hash"